src/hotspot/share/opto/library_call.cpp

```diff
@@ -5206,24 +5206,102 @@ bool LibraryCallKit::inline_vectorizedMismatch() {
   const TypeAryPtr* top_a = a_type->isa_aryptr();
   const TypeAryPtr* top_b = b_type->isa_aryptr();
   if (top_a == NULL || top_a->klass() == NULL ||
-    top_b == NULL || top_b->klass() == NULL) {
+      top_b == NULL || top_b->klass() == NULL) {
     // failed array check
     return false;
   }
-
-  Node* call;
   jvms()->set_should_reexecute(true);
 
   Node* obja_adr = make_unsafe_address(obja, aoffset);
   Node* objb_adr = make_unsafe_address(objb, boffset);
 
-  call = make_runtime_call(RC_LEAF,
-    OptoRuntime::vectorizedMismatch_Type(),
-    stubAddr, stubName, TypePtr::BOTTOM,
-    obja_adr, objb_adr, length, scale);
+  bool enable_pi = false;
+  Node* fast_path = top();
+  Node* fastcomp_result = top();
+  Node* init_mem = map()->memory();
+
+  assert(scale->bottom_type()->isa_int(), "scale must be integer");
+  if (scale->bottom_type()->is_int()->is_con()) {
+    int scale_val = scale->bottom_type()->is_int()->get_con();
+    BasicType prim_types[] = {T_BYTE, T_SHORT, T_INT, T_LONG};
+    BasicType elem_bt = prim_types[scale_val];
+    int vec_len = ArrayOperationPartailInlineSize / type2aelembytes(elem_bt);
+
+    // Enable partial in-lining if compare size is less than
+    // ArrayOperationPartailInlineSize(default 32 bytes).
+    // If ArrayOperationPartailInlineSize > 32 inlining is enabled
+    // for all integral types (byte/short/char/int), else for default
+    // value inlining is enabled for subword types (byte/short/char).
+    if (ArrayOperationPartailInlineSize > 32) {
+      enable_pi = is_subword_type(elem_bt) || elem_bt == T_INT;
+    } else if (ArrayOperationPartailInlineSize) {
```

Comment:
Should this be a comparison against something? "if (ArrayOperationPartailInlineSize) {"